cmd/api-mock/main.go

```diff
@@ -30,9 +36,55 @@ func Health(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+type ClustersEndpoint struct {
+	data    map[ClustersKey]string
+	dataMux *sync.RWMutex
+}
+
+func NewClustersEndpoint() *ClustersEndpoint {
+	return &ClustersEndpoint{
+		data: map[ClustersKey]string{},
+	}
+}
+
+type ClustersKey struct {
+	ExternalId string
+}
+
+func (cs *ClustersEndpoint) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+
+	case http.MethodGet:
+		cs.dataMux.RLock()
+		defer cs.dataMux.RUnlock()
+
+		search := r.URL.Query().Get("search")
+
+		//for the mock server, we don't care about the expression itself,
+		//we just want the cluster external id out of it
+		//e.g.: external_id = 'a440b136-b2d6-406b-a884-fca2d62cd170'
+		//get the id, with quotes
+		re := regexp.MustCompile(`'.*'`)
```

Comment:
nit, could be declared outside of this function, so the regex is not recompiled for every http call.